gnome-image-installer/util/gis-unattended-config.c

```diff
@@ -282,6 +332,44 @@ gis_unattended_config_get_locale (GisUnattendedConfig *self)
   return self->locale;
 }
 
+/**
+ * gis_unattended_config_get_image:
+ *
+ * Returns: the basename of the configured image, or %NULL if no specific image
+ *  is configured.
+ */
+const gchar *
+gis_unattended_config_get_image (GisUnattendedConfig *self)
+{
+  return self->filename;
+}
+
+/**
+ * gis_unattended_config_matches_device:
+ * @device: full path to a block device
+ *
+ * Returns: %TRUE if @device matches the configured target device, or if none
+ *  is configured.
+ */
+gboolean
+gis_unattended_config_matches_device (GisUnattendedConfig *self,
+                                      const gchar *device)
+{
+  const gchar *basename;
+
+  if (self->blockdevice == NULL)
+    return TRUE;
+
+  if (self->blockdevice[0] == '/')
+    return g_strcmp0 (device, self->blockdevice) == 0;
+
+  basename = rindex (device, '/');
```

Comment:
Nitpick: It’s not really the basename if you get a pointer to the final `/`, it’s the basename plus a leading `/`.

Why not use `g_path_get_basename()`?

Reply:
Fair.